src/main/java/io/jenkins/plugins/ml/FileParser.java

```diff
@@ -0,0 +1,143 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildWrapper;
+import hudson.tasks.BuildWrapperDescriptor;
+import io.jenkins.plugins.ml.model.ParsableFile;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class FileParser extends BuildWrapper {
+    public   List<ParsableFile> parsableFiles;
+    private List<String> copiedFiles = new ArrayList<>();
+
+    private static final Logger LOGGER = Logger.getLogger(FileParser.class.getName());
+
+    @DataBoundConstructor
+    public FileParser(ArrayList<ParsableFile> parsableFiles){
+        this.parsableFiles = parsableFiles;
+    }
+    @DataBoundSetter
+    public void setParsableFiles(List<ParsableFile> parsableFiles){
+        this.parsableFiles = parsableFiles;
+    }
+
+    @Override
+    public Environment setUp(AbstractBuild build, final Launcher launcher,
+                             BuildListener listener) throws IOException, InterruptedException
+    {
+        // Get the workspace
+        LOGGER.info("Recognize project workspace and folder");
+        FilePath projectWorkspace = build.getWorkspace();
+
+        // Get remote workspace if local workspace is null
+        if(projectWorkspace == null && build.getWorkspace().getRemote() != null){
+            projectWorkspace = new FilePath(new File((build.getWorkspace()).getRemote()));
+        }
+
+        // Copy each file to the workspace
+        for (ParsableFile file :  parsableFiles) {
+            FilePath copyFrom = new FilePath(new File(file.getFilePath()));
+
+
+            LOGGER.info("Copying file from " + copyFrom.toURI()
+                    + " to " + projectWorkspace.toURI());
+            String copyToFile = file.getFilePath().substring(file.getFilePath().lastIndexOf("/")+1);
```

Comment:
Is it equivalent to `String copyToFile = copyFrom.getName();` ?

Reply:
Oh yes. `copyFrom` is FilePath right. I could have done it easily. will change it.